engine/execution/computation/computer/computer.go

```diff
@@ -321,10 +321,13 @@ func (e *blockComputer) executeTransaction(
 ) error {
 	startedAt := time.Now()
 
+	sample := []metrics.Sample{{Name: "/gc/heap/allocs:bytes"}}
+	metrics.Read(sample)
+
 	var memAllocBefore uint64
-	var m runtime.MemStats
-	runtime.ReadMemStats(&m)
-	memAllocBefore = m.TotalAlloc
+	if sample[0].Value.Kind() == metrics.KindUint64 {
+		memAllocBefore = sample[0].Value.Uint64()
```

Comment:
too much copy/paste ... refactor this into a function.  maybe stick it into util/debug?